src/OpenApi/src/Services/OpenApiDocumentService.cs

```diff
@@ -296,11 +296,78 @@ private OpenApiRequestBody GetFormRequestBody(IList<ApiRequestFormat> supportedR
+        // bind a single Todo model instance to the todo parameter. Grouping by name allows us to handle both cases.
+        var groupedFormParameters = formParameters.GroupBy(parameter => parameter.ParameterDescriptor.Name).ToList();
+        // If there is only one real parameter derive from the form body, then set it directly in the schema.
+        if (groupedFormParameters is [var groupedParameter])
         {
-            schema.Properties[parameter.Name] = _componentService.GetOrCreateSchema(parameter.Type);
+            // If the group contains one element, then we're dealing with a minimal API scenario where a
+            // single form parameter produces a single API description parameter.
+            if (groupedParameter.Count() == 1)
+            {
+                var description = groupedParameter.Single();
+                // Form files are keyed by their parameter name so we must capture the parameter name
+                // as a property in the schema.
+                if (description.Type == typeof(IFormFile) || description.Type == typeof(IFormFileCollection))
+                {
+                    schema.Properties[description.Name] = _componentService.GetOrCreateSchema(description.Type);
+                }
+                else
+                {
+                    // POCOs do not need to be subset under their parameter name. The form-binding implementation
+                    // will capture them implicitly.
+                    schema = _componentService.GetOrCreateSchema(description.Type);
+                }
+            }
+            // If there are multiple API description parameters in the group, then we are dealing
+            // with the MVC scenario where form parameters are exploded into separate API parameters
+            // for each property within the complex type.
+            else
+            {
+                foreach (var parameter in groupedParameter)
+                {
+                    schema.Properties[parameter.Name] = _componentService.GetOrCreateSchema(parameter.Type);
+                }
+            }
+        }
+        // If there are multiple parameters sourced from the form, then we use `allOf` to capture each parameter.
+        else
+        {
+            // Process the arguments in the same way that we do for the single-parameter case but
+            // set each sub-schema as an `allOf` item in the schema.
+            foreach (var parameter in groupedFormParameters)
+            {
+                if (parameter.Count() == 1)
+                {
+                    var description = parameter.Single();
+                    if (description.Type == typeof(IFormFile) || description.Type == typeof(IFormFileCollection))
+                    {
+                        schema.AllOf.Add(new OpenApiSchema { Type = "object", Properties = new Dictionary<string, OpenApiSchema> { [description.Name] = _componentService.GetOrCreateSchema(description.Type) } });
```

Comment:
This feels like it wants to be multiple lines. 😝

Reply:
Using an ultra-wide monitor has me living in an alternate dimension. 😆 

I'll apply a format here to break it up.